src/libmain.c

```diff
@@ -765,16 +769,20 @@ static gint setup_config_dir(void)
 	return mkdir_result;
 }
 
-/* Signal handling removed since main_quit() uses functions that are
- * illegal in signal handlers
-static void signal_cb(gint sig)
+
+#ifdef G_OS_UNIX
+static gboolean signal_cb(gpointer user_data)
 {
+	gint sig = GPOINTER_TO_INT(user_data);
 	if (sig == SIGTERM)
 	{
+		geany_debug("Received SIGTERM signal");
 		main_quit();
 	}
+	return G_SOURCE_REMOVE;
```

Comment:
I guess it depends on whether or not returning `CONTINUE` will remove the signal handler altogether and restore the default action or not.

Because IMO
* it makes more sense to return `CONTINUE`, unless we *mean* not to handle the signal anymore
* it is often customary to stop handling these kind of signals after receiving them for the first time, so that if the app still doesn't quit sending it again will just terminate it, just not necessarily cleanly.  So that might suggest `REMOVE`.